pkg/custom_detectors/custom_detectors.go

```diff
@@ -82,66 +83,98 @@ func (c *customRegexWebhook) FromData(ctx context.Context, verify bool, data []b
 	// ]
 	matches := permutateMatches(regexMatches)
 
+	g := new(errgroup.Group)
+
 	// Create result object and test for verification.
+	resultsCh := make(chan detectors.Result, maxTotalMatches)
 	for _, match := range matches {
+		match := match
+		g.Go(func() error {
+			err := c.createResults(ctx, match, verify, resultsCh)
+			return err
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+	close(resultsCh)
```

Comment:
suggestion: I think we should move this up right after you created the chan and defer the close. Otherwise if we encounter an err in the goroutines w/in the errgroup we are going to leak with this unclosed channel.

so just: `defer close(resultsCh)` at L90

Reply:
aaah! yes that's way cleaner!!!

Reply:
Unfortunately we need to `close` the channel before iterating over it otherwise the loop on line 103 would never end.

I do think that maybe we can ignore the error returned by `g.Wait()` though, and collect as many results that we got.